### PR TITLE
Add NNCF optimization for DeepFloyd IF notebook

### PR DESCRIPTION
**Pull request by l-bat**
Add NNCF optimization for DeepFloyd IF model from [238-deep-floyd-if.ipynb](https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/238-deeployd-if/238-deep-floyd-if.ipynb)

### CODE REVIEW

**Comment by l-bat** [comment]
@MaximProshin, @eaidova please take a look

File: notebooks/238-deeployd-if/README.md [hunk 1]
```diff
@@ -17,6 +17,14 @@ The result of notebook work demonstrated on the image below:
 
 >**Note**: Please be aware that a machine with at least 32GB of RAM is necessary to run this example.
 
+The [second notebook](238-deep-floyd-if-optimize.ipynb) is about the optimization by 8-bit post-training quantization and weights compression and consists of the following steps:
+1. Compress weights of the converted OpenVINO text encoder from [notebook](238-deep-floyd-if.ipynb) with NNCF.
```
Comment (MaximProshin) [inline notebooks/238-deeployd-if/README.md:21]: let's change 'notebook' to 'first notebook' everywhere in this list and in 238-deep-floyd-if-optimize.ipynb

**Comment by MaximProshin** [comment]
I still plan to run the notebook myself and will provide more feedback later

**Comment by MaximProshin** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1331/discussion/#comment-1727118254) on ReviewNB
> 
> _MaximProshin commented on 2023-09-20T07:23:29Z_ ----------------------------------------------------------------
> 
> Can we somehow run it as one step and then provide a summary? At this moment it's really hard to catch the difference and one needs to scroll a lot to compare models.

One option could be to reduce the output of benchmark_app tool and show only fps. I guess it's possible.

**Comment by eaidova** [comment]
> > View / edit / reply to [this conversation](https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1331/discussion/#comment-1727118254) on ReviewNB
> > _MaximProshin commented on 2023-09-20T07:23:29Z_ ----------------------------------------------------------------
> > Can we somehow run it as one step and then provide a summary? At this moment it's really hard to catch the difference and one needs to scroll a lot to compare models.
> 
> One option could be to reduce the output of benchmark_app tool and show only fps. I guess it's possible.

I think simple solution from this notebook, can be reused here for parsing and representing benchmark logs https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/112-pytorch-post-training-quantization-nncf/112-pytorch-post-training-quantization-nncf.ipynb

**Comment by MaximProshin** [comment]
the folder name is "238-deeployd-if" missing 'f'. I wonder if we could or should fix it here. @eaidova , what do you think?

**Comment by eaidova** [comment]
> the folder name is "238-deeployd-if" missing 'f'. I wonder if we could or should fix it here. @eaidova , what do you think?

@MaximProshin, I think we (ov notebooks team) should fix it on our side (as it is not related to this notebook scenario), thanks for note

**Comment by l-bat** [comment]
<p>Callback only can be used together with <code>start_async</code>, but <code>UnetFirstStage </code>and <code>UnetSecondStage</code>use __call__ instead of it.&nbsp;</p><p>On the other hand, using start_async for these classes is redundant since it's only need synchronious call. </p><p>I rewrote wrapper to reduce code size.</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1331/discussion/#comment-1727146284'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: notebooks/238-deeployd-if/README.md (hunk 1 above)
Comment (l-bat) [reply notebooks/238-deeployd-if/README.md]: done

Comment (MaximProshin) [reply notebooks/238-deeployd-if/README.md]: please also fix it in 238-deep-floyd-if-optimize.ipynb

**Review by MaximProshin — APPROVED** [review]

**Comment by l-bat** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1331/discussion/#comment-1731321923'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by eaidova — APPROVED** [review]

**Pull request merged**